src/Broadway/EventStore/MongoDBEventStore.php

```diff
@@ -0,0 +1,367 @@
+     * @param string                                     $aggregateId
+     *
+     * @return array
+     */
+    private function extractEvents(DomainEventStreamInterface $eventStream, $aggregateId)
+    {
+        $events = array();
+
+        foreach ($eventStream as $message) {
+            $this->assertIdentifier($aggregateId, $message->getId());
+
+            $events[] = $this->serializeEvent($message);
+        }
+
+        return $events;
+    }
+
+    /**
+     * @param string $expected
+     * @param string $actual
+     */
+    private function assertIdentifier($expected, $actual)
+    {
+        if (0 != strcmp($expected, $actual)) {
+            throw new InvalidIdentifierException(sprintf('Expecting %s, got %s identifier.', $expected, $actual));
+        }
+    }
+
+    /**
+     * @param DomainMessage $message
+     *
+     * @return array
+     */
+    private function serializeEvent(DomainMessage $message)
+    {
+        $data = array(
+            self::FIELD_STREAM_ID   => (string) $message->getId(),
+            self::FIELD_PLAYHEAD    => $message->getPlayhead(),
+            self::FIELD_METADATA    => $this->metadataSerializer->serialize($message->getMetadata()),
+            self::FIELD_PAYLOAD     => $this->payloadSerializer->serialize($message->getPayload()),
+            self::FIELD_RECORDED_ON => $message->getRecordedOn()->toString(),
+            self::FIELD_TYPE        => $message->getType(),
+        );
+
+        // save space
+        if (empty($data[self::FIELD_METADATA])) {
+            unset($data[self::FIELD_METADATA]);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $row
+     *
+     * @return DomainMessage
+     */
+    private function deserializeEvent(array $row)
+    {
+        if (empty($row[self::FIELD_METADATA])) {
+            $metadata = new Metadata();
+        } else {
+            $metadata = $this->metadataSerializer->deserialize($row[self::FIELD_METADATA]);
+        }
+
+        return new DomainMessage(
+            $row[self::FIELD_STREAM_ID],
+            $row[self::FIELD_PLAYHEAD],
+            $metadata,
+            $this->payloadSerializer->deserialize($row[self::FIELD_PAYLOAD]),
+            DateTime::fromString($row[self::FIELD_RECORDED_ON])
+        );
+    }
+
+    /**
+     * @param array  $event
+     * @param string $commitId
+     */
+    private function saveEvent(array $event, $commitId = null)
+    {
+        if ($commitId) {
+            $event[self::FIELD_COMMIT_ID] = $commitId;
+        }
+
+        $this->eventCollection->insert($event, array(
+            'safe' => true,
+        ));
+    }
+
+    /**
+     * @param array $events
+     *
+     * @throws Exception
+     * @throws MongoException
+     */
+    private function saveBulkEvents(array $events)
+    {
+        $commitId = $this->getNewId();
+
+        try {
+            $this->beginTransaction($commitId);
+
+            foreach ($events as $event) {
```

Comment:
Why not use [batchInsert](http://php.net/manual/en/mongocollection.batchinsert.php) here?